pkg/payload/payload_test.go

```diff
@@ -149,39 +152,83 @@ func Test_Exclude(t *testing.T) {
 			isExcluded: true,
 		},
 		{
+			name:        "profile-selection-works",
 			profile:     "single-node",
 			annotations: map[string]interface{}{"include.release.openshift.io/self-managed-high-availability": "true"},
 			isExcluded:  true,
 		},
 		{
+			name:        "profile-selection-works-included",
 			profile:     DefaultClusterProfile,
 			annotations: map[string]interface{}{"include.release.openshift.io/self-managed-high-availability": "true"},
 		},
 		{
```

Comment:
Add two test cases [for][1]:

> If the value is not "TechPreviewNoUpgrade", the manifest is never created.

where you set the annotation to `DoesNotExist` and show that it's excluded regardless if `includeTechPreview`?

[1]: https://github.com/openshift/enhancements/pull/957/files#diff-6f9f82c1e111102d550554343fc634b0908acf22a0aa1490c0801f7ddf85b213R61

Reply:
Ah, you have these for `Other` already, but your exclusion sense on those tests diverges from your enhancement plan.

Reply:
fixed in fda77fe; thanks :)